cupyx/scipy/interpolate/_polyint.py

```diff
@@ -0,0 +1,323 @@
+        """
+        Actually evaluate the value of the interpolator
+        """
+        raise NotImplementedError()
+
+    def _prepare_x(self, x):
+        """
+        Reshape input array to 1-D
+        """
+        x = _asarray_validated(x, check_finite=False, as_inexact=True)
+        x_shape = x.shape
+        return x.ravel(), x_shape
+
+    def _finish_y(self, y, x_shape):
+        """
+        Reshape interpolated y back to an N-D array similar to initial y
+        """
+        y = y.reshape(x_shape + self._y_extra_shape)
+        if self._y_axis != 0 and x_shape != ():
+            nx = len(x_shape)
+            ny = len(self._y_extra_shape)
+            s = (list(range(nx, nx + self._y_axis))
+                 + list(range(nx)) + list(range(nx + self._y_axis, nx + ny)))
+            y = y.transpose(s)
+        return y
+
+    def _reshape_yi(self, yi, check=False):
+        """
+        Reshape the updated yi to a 1-D array
+        """
+        yi = cupy.moveaxis(yi, self._y_axis, 0)
+        if check and yi.shape[1:] != self._y_extra_shape:
+            ok_shape = "%r + (N,) + %r" % (self._y_extra_shape[-self._y_axis:],
+                                           self._y_extra_shape[:-self._y_axis])
+            raise ValueError("Data must be of shape %s" % ok_shape)
+        return yi.reshape((yi.shape[0], -1))
+
+    def _set_yi(self, yi, xi=None, axis=None):
+        if axis is None:
+            axis = self._y_axis
+        if axis is None:
+            raise ValueError("no interpolation axis specified")
+
+        shape = yi.shape
+        if shape == ():
+            shape = (1,)
+        if xi is not None and shape[axis] != len(xi):
+            raise ValueError("x and y arrays must be equal in length along "
+                             "interpolation axis.")
+
+        self._y_axis = (axis % yi.ndim)
+        self._y_extra_shape = yi.shape[:self._y_axis]+yi.shape[self._y_axis+1:]
+        self.dtype = None
+        self._set_dtype(yi.dtype)
+
+    def _set_dtype(self, dtype, union=False):
+        if cupy.issubdtype(dtype, cupy.complexfloating) \
+                or cupy.issubdtype(self.dtype, cupy.complexfloating):
+            self.dtype = cupy.complex_
+        else:
+            if not union or self.dtype != cupy.complex_:
+                self.dtype = cupy.float_
+
+
+class BarycentricInterpolator(_Interpolator1D):
+    """The interpolating polynomial for a set of points.
+
+    Constructs a polynomial that passes through a given set of points.
+    Allows evaluation of the polynomial, efficient changing of the y
+    values to be interpolated, and updating by adding more x values.
+    For reasons of numerical stability, this function does not compute
+    the coefficients of the polynomial.
+
+    The value `yi` need to be provided before the function is
+    evaluated, but none of the preprocessing depends on them,
+    so rapid updates are possible.
+
+    Parameters
+    ----------
+    xi : array-like
```

Comment:
```suggestion
    xi : cupy.ndarray
```

Reply:
Please change the rest of docstrings (sorry, I messed up the line)

Reply:
Sure, fixed all. Thanks!

(I'll push commit after addressing all the reviews :))